### PR TITLE
Convert scala to java in application module

### PR DESCRIPTION
**Pull request by ollivir**
A previous version had a flawed conversion of the builder_cannot_be_reused() test in ApplicationBuilderTest which did not clean up the initialised application container. This caused any next test case using the container to fail.

### CODE REVIEW

File: application/src/test/java/com/yahoo/application/ApplicationBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        } catch (RuntimeException e) {
+            assertThat(e.getMessage(), containsString("build method"));
+        }
+    }
+
+    private interface TestCase {
+        public void accept(ApplicationBuilder ab) throws Exception;
+    }
+
+    public void withApplicationBuilder(TestCase f) throws Exception {
```
Comment (gjoranv) [inline application/src/test/java/com/yahoo/application/ApplicationBuilderTest.java]: Declare helper method private

File: application/src/test/java/com/yahoo/application/ApplicationBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        try (Application app = builder.build()) {
+            builder.servicesXml("");
+            fail("Expected exception.");
+        } catch (RuntimeException e) {
+            assertThat(e.getMessage(), containsString("build method"));
+        }
+    }
+
+    private interface TestCase {
+        public void accept(ApplicationBuilder ab) throws Exception;
```
Comment (gjoranv) [inline application/src/test/java/com/yahoo/application/ApplicationBuilderTest.java]: Remove redundant 'public'

**Review by gjoranv — CHANGES_REQUESTED** [review]

File: application/src/test/java/com/yahoo/application/ApplicationBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        builder.servicesXml("<jdisc version=\"1.0\" />");
+        try (Application application = builder.build()) {
+            // do nothing
+        }
+
+        builder.servicesXml(""); // should fail
+    }
+
+    private interface TestCase {
+        public void accept(ApplicationBuilder ab) throws Exception;
```
Comment (gjoranv) [inline application/src/test/java/com/yahoo/application/ApplicationBuilderTest.java]: Remove redundant 'public'

**Comment by ollivir** [comment]
Changes made according to comments (I also made the helper method static)

**Review by gjoranv — APPROVED** [review]

**Pull request merged**